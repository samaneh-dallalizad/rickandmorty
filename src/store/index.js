// Create a new store instance.
import {createStore} from "vuex";

export const store = createStore({
    state () {
        return {
            characters: []
        }
    },
    mutations: {
        setCharacters (state, characters) {
            state.characters = characters.slice(0,10)
        }
    },
    actions: {
        async getCharacters ({commit}) {
            let response = await fetch("https://rickandmortyapi.com/api/character")
            let data = await response.json()
            commit("setCharacters", data["results"])
        }
    }
})
