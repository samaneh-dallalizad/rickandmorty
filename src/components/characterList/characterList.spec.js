import { expect, it, describe } from 'vitest'
import Vue from 'vue';
import {mount} from '@vue/test-utils'
import CharacterListComponent from './characterList.vue';
import {store} from '../../store';
import {fetch} from 'whatwg-fetch'

describe('HomepageComponent', () => {

    it('CharacterList without data must always be the same', async () => {
        const wrapper = mount(CharacterListComponent, {
            global: {
                provide: {
                    store
                },
                fetch
            }
        })
        expect(wrapper.element).toMatchSnapshot()
    })

    it('CharacterList with data must always be the same', async () => {
        await store.dispatch('getCharacters')
        const wrapper = mount(CharacterListComponent, {
            global: {
                provide: {
                    store
                },
                fetch
            }
        })
        expect(wrapper.element).toMatchSnapshot()
    })
})