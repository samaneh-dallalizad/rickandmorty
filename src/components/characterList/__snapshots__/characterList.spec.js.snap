// Vitest Snapshot v1

exports[`HomepageComponent > CharacterList with data must always be the same 1`] = `
<div
  data-v-app=""
>
  
  <!--if characters more than 0-->
  <section
    data-v-5fecca80=""
  >
    <!--details-->
    
    <!--modal-->
    <div
      class="modal"
      data-v-5fecca80=""
    >
      <div
        class="modal-background"
      />
      <div
        class="modal-card"
      >
        <!--header-->
        <header
          class="modal-card-head"
        >
          <!--title/character_name-->
          <p
            class="modal-card-title"
          >
            character: 
          </p>
          <!--close button-->
          <button
            aria-label="close"
            class="button is-delete is-danger delete"
          >
            Close 
          </button>
        </header>
        <!--body/content-->
        <section
          class="modal-card-body"
        >
          <div
            class="columns"
          >
            <!--character_image-->
            <div
              class="column is-4"
            >
              <figure
                class="image is-128x128 mx-auto mb-10px"
              >
                <img
                  alt="{{ character?.origin.name }}"
                  class="is-rounded"
                />
              </figure>
            </div>
            <div
              class="column is-8"
            >
              <ul>
                <!--character_Origin-->
                <li>
                  <strong
                    class="is-inline has-text-primary"
                  >
                     Origin: 
                  </strong>
                   
                </li>
                <!--character_status-->
                <li>
                  <strong
                    class="is-inline has-text-primary"
                  >
                    Status: 
                  </strong>
                   
                </li>
                <!--character_gender-->
                <li>
                  <strong
                    class="is-inline has-text-primary"
                  >
                     gender: 
                  </strong>
                   
                </li>
                <!--character_species-->
                <li>
                  <strong
                    class="is-inline has-text-primary"
                  >
                     species: 
                  </strong>
                   
                </li>
                <!--character_lacation-->
                <li>
                  <strong
                    class="is-inline has-text-primary"
                  >
                     location: 
                  </strong>
                   
                </li>
              </ul>
            </div>
          </div>
        </section>
      </div>
    </div>
    
    <!--list-->
    <h1
      class="has-text-warning mb-6 has-text-centered is-family-monospace"
      data-v-5fecca80=""
    >
       Rick and Morty 
    </h1>
    <!--container-->
    <div
      class="container"
      data-v-5fecca80=""
    >
      <!--columns-->
      <ul
        class="columns is-multiline"
        data-v-5fecca80=""
      >
        
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Rick Sanchez
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    51
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Morty Smith
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    51
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Summer Smith
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    42
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Beth Smith
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    42
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Jerry Smith
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    39
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Abadango Cluster Princess
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    1
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Abradolf Lincler
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    2
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Adjudicator Rick
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    1
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Agency Director
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    1
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          class="column is-3"
          data-v-5fecca80=""
        >
          <!--card-->
          <div
            class="card p-0 is-clickable"
            data-v-5fecca80=""
          >
            <!--card image-->
            <div
              class="card-image"
              data-v-5fecca80=""
            >
              <figure
                class="image is-4by3"
                data-v-5fecca80=""
              >
                <img
                  data-v-5fecca80=""
                  src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
                />
              </figure>
            </div>
            <!--card content-->
            <div
              class="card-content"
              data-v-5fecca80=""
            >
              <ul
                data-v-5fecca80=""
              >
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    Alan Rails
                  </strong>
                </li>
                <li
                  data-v-5fecca80=""
                >
                  <strong
                    class="has-text-white"
                    data-v-5fecca80=""
                  >
                    episodes:
                  </strong>
                   
                  <span
                    class="tag has-background-primary mt-4"
                    data-v-5fecca80=""
                  >
                    1
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </li>
        
      </ul>
    </div>
  </section>
  
</div>
`;

exports[`HomepageComponent > CharacterList without data must always be the same 1`] = `
<div
  data-v-app=""
>
  
  <!--if characters more than 0-->
  
  <!--if characters more than 0-->
  <section
    class="notification is-warning is-fullwidth"
    data-v-5fecca80=""
  >
    OOps! some thing wrong has happened
  </section>
  
  
</div>
`;
